src/Facebook/InstantArticles/Transformer/Transformer.php

```diff
@@ -231,27 +185,12 @@ public function transform($context, $node)
                 }
                 $matched = false;
 
-                // Get all classes and interfaces this context extends/implements
```

Comment:
Why are we removing the performance optimization? The algorithm is 10 times slower without it.

Reply:
The changes you reverted on addRule, getRules and getAllClassTypes were all related to this optimization.

Reply:
Here's the original perf PR for reference: #46

Reply:
This is one of the discussion points.
This was all broken(after strict migration), not well tested and documented. Got rid of it.
We can add it back with proper review.